agreement_dynamic/views/agreement.xml

```diff
@@ -0,0 +1,199 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_agreement_form" model="ir.ui.view">
+        <field name="name">agreement.form.view</field>
+        <field name="model">agreement</field>
+        <field name="inherit_id" ref="agreement.agreement_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='name']" position="after">
+                <h2 attrs="{'invisible': [('is_template', '=', False)]}">
+                    <label for="model_name" />
+                    <field name="model_name" />
+                </h2>
+            </xpath>
+            <!-- remove signature date from form view...-->
+            <xpath expr="//field[@name='signature_date']" position="replace">
+            </xpath>
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button
+                    name="action_toggle_active"
+                    type="object"
+                    class="oe_stat_button"
+                    icon="fa-archive"
+                >
+                    <field
+                        name="active"
+                        widget="boolean_button"
+                        options="{'terminology': {
+                            'string_true': 'ACTIVE',
+                            'hover_true': 'PAUSE',
+                            'string_false': 'HELP',
+                            'hover_false': 'FINISH'
+                        }}"
+                    />
+                </button>
+                <field name="section_ids" invisible="1" />
+                <button
+                    type="object"
+                    name="action_view_section"
+                    class="oe_stat_button"
+                    icon="fa-file-text-o"
+                    attrs="{'invisible': [('section_count', '=', 0)]}"
+                    groups="base.group_user"
+                >
+                    <field name="section_count" widget="statinfo" string="Sections" />
+                </button>
+            </xpath>
+            <xpath expr="//field[@name='is_template']" position="after">
+                <field
+                    name="template_id"
+                    attrs="{'readonly': [('signature_date', '!=', False)]}"
+                />
+                <!--
+                editable only for templates
+                 -->
+                <field
+                    name="model_id"
+                    attrs="{'invisible':[('is_template', '=', False)]}"
+                />
+                <label for="resource_ref" />
+                <div>
+                    <field
+                        name="resource_ref"
+                        options="{'hide_model': True, 'no_create': True, 'no_edit': True, 'no_open': True}"
+                        class="oe_inline"
+                    />
```

Comment:
For some reason when I select 'Country' as a model, and then I switch to `hr.contract`, and I save, this field becomes hidden (I think because there are no records in Contract)

If I dont save, it becomes all messy because it will still show records of Country, even I have switched already to Contract.

Not sure what to do with that

Reply:
Yes, this is a problem I encountered myself too, when trying to split resource_ref into two separate fields.
When  there are no records, or if the res_id part of resource_ref is not filled manually, then this whole thing becomes read only permanently.

For that reason, in the _compute for resource_ref I do not check if res_id is filled. If it is filled, I use that value. If it is not filled, then we pick a random record from resource_ref model_id as a placeholder.

So when we have a template, we can edit both of model_id and res_id freely. When we create a report from that template, then model_id becomes readonly and user can select records of model_id from the dropdown. Template editor should be aware of this limitation in templates then. If he/she is, this is fine.